Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -922,6 +923,12 @@ extension SwiftLanguageService {
     return Array(hints)
   }
 
+  public func codeLens(_ req: CodeLensRequest) async throws -> [CodeLens] {
+    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    let syntaxTreeManager = SyntaxTreeManager()
```

Comment:
We should use the shared `syntaxTreeManager` on `SwiftLanguageService`. That way we don’t need to re-parse the syntax tree if the source file already has a parsed syntax tree (which it most likely does).